src/transform/ReferenceCollection.js

```diff
@@ -0,0 +1,253 @@
+import ElementUtilities from './ElementUtilities'
+import NodeUtilities from './NodeUtilities'
+import Polyfill from './Polyfill'
+
+const REFERENCE_SELECTOR = '.reference'
+
+/**
+ * Is Citation.
+ * @param  {!string}  href
+ * @return {!boolean}
+ */
+const isCitation = href => href.indexOf('#cite_note') > -1
+
+/**
+ * Gets first child anchor.
+ * @param  {!HTMLElement} element
+ * @return {?HTMLAnchorElement}
+ */
+const getFirstChildAnchor = element => element.querySelector('A')
+
+/**
+ * Determines if node is a text node containing only whitespace.
+ * @param  {!Node}  node
+ * @return {!boolean}
+ */
+const isWhitespaceTextNode = node =>
+  !(!node || node.nodeType !== Node.TEXT_NODE || !node.textContent.match(/^\s+$/))
+
+/**
+ * Checks if element has a child anchor with a citation link.
+ * @param  {!HTMLElement}  element
+ * @return {!boolean}
+ */
+const hasCitationLink = element => {
+  try {
+    return isCitation(getFirstChildAnchor(element).getAttribute('href'))
+  } catch (e) {
```

Comment:
Hmm I don't recall the exact reason for the `try/catch` - been a while. I played with your alternative for a bit (adding call to `isCitation` where needed) but didn't get it to work (tests failed). Would be ok to leave this as-is for now?

Reply:
Oh nevermind I think I fixed it...

Reply:
Perfect! I'm glad we have a test case that fails! Now we can easily make the improvement right now and be confident in our code instead of punting it down the line and be confused later.

I looked into this myself. This try / catch block was obscuring the real underlying issue which was that adjacentNonWhitespaceNode() can return null and getFirstChildAnchor() expects nonnull.

Reply:
Heya just a heads-up I reverted your change - see my comment here https://github.com/wikimedia/wikimedia-page-library/pull/140#issuecomment-396567201 for details.